src/beamlime/applications/handlers.py

```diff
@@ -143,23 +153,55 @@ def update_histogram(self, message: WorkflowResultUpdate) -> None:
         for name, data in content.items():
             self.plot_item(name, data)
 
+    def show(self):
+        """Show the figures in a grid layout."""
+        from plopp.widgets import Box, HBar, VBar
+
+        figures = list(self.figures.values())
+        n_rows = len(figures) // self.max_column + 1
+        return Box(
+            VBar(
+                [
+                    HBar(
+                        figures[
+                            i_row * self.max_column : i_row * self.max_column
+                            + self.max_column
+                        ]
+                    )
+                    for i_row in range(n_rows)
+                ]
+            )
+        )
+
 
 class PlotSaver(PlotStreamer):
     """Plot handler to save the updated histogram into an image file."""
 
-    def __init__(self, logger: BeamlimeLogger, image_path_prefix: ImagePath) -> None:
-        super().__init__(logger)
+    def __init__(
+        self,
+        *,
+        logger: BeamlimeLogger,
+        image_path_prefix: ImagePath,
+        max_column: MaxPlotColumn = DefaultMaxPlotColumn,
+    ) -> None:
+        super().__init__(logger=logger, max_column=max_column)
         self.image_path_prefix = image_path_prefix
 
     def save_histogram(self, message: WorkflowResultUpdate) -> None:
+        from plopp import tiled
+
         super().update_histogram(message)
-        self.info("Received histogram, saving into %s...", self.image_path_prefix)
-        for name, figure in self.figures.items():
-            figure.save(f'{self.image_path_prefix}-{name}.png')
+        image_file_name = f"{self.image_path_prefix}.png"
+        self.info("Received histogram(s), saving into %s...", image_file_name)
+        plot_tile = tiled((len(self.figures) // self.max_column) + 1, self.max_column)
+        for i_fig, figure in enumerate(self.figures.values()):
+            plot_tile[i_fig // self.max_column, i_fig % self.max_column] = figure
+
+        plot_tile.save(image_file_name)
```

Comment:
I probably missed that discussion, but when using this from the command line, why are we saving png files instead of opening a qt window which would show the matplotlib figure? Or maybe we are doing that somewhere else and that should also be updated?

Reply:
The default visualisation should do bare-minimum I thought.
Dumping a image file was the minimum since we can just use any viewers, without any dependencies of GUI libraries.
So I didn't really think about having QT window here.
Do you think we should...?

Reply:
I guess what I meant was calling Matplotlib's `show` on a figure and it opens the window for you.

Reply:
How should we do that...? 
I tried simply calling `plt.show()` but it doesn't do anything in the terminal though...

Reply:
Maybe it's because you're in VScode? https://stackoverflow.com/a/65086012

Reply:
Probably.
I think the default Visualisation should work in any environment,
so if it requires any sort of effort, I'd go for dumping a file.

Also, I'm not sure how to only-show the grid-view of all plots, not the individual ones.

Reply:
Have you tried just from a terminal to make sure that this is indeed the issue? If it is, then I agree with you.

> Also, I'm not sure how to only-show the grid-view of all plots, not the individual ones.

Not sure I understood exactly what you meant here...

Reply:
> Have you tried just from a terminal to make sure that this is indeed the issue? If it is, then I agree with you.

I just tried, and it didn't pop up the plot... 
it only works when I created plot directly using `pyplot`.
And whenever the window pops up, it blocks the process, and if I use `block=False` option, it just doesn't show up at all... 

> > Also, I'm not sure how to only-show the grid-view of all plots, not the individual ones.
> 
> Not sure I understood exactly what you meant here...

I just don't know how to update the plot display with the new one, without making a new window...
And if you plot sth with `pyplot` and then use `plt.show()`, it will show all the plots you have drawn before. But we only want one grid plots, without the individual ones.

Reply:
> it only works when I created plot directly using pyplot.

Yeah, it seems there is something wrong with getting figures to show from the terminal. I am getting this error
```
AttributeError: Figure.show works only for figures managed by pyplot, normally created by pyplot.figure()
```
Is that also what you got?

I opened https://github.com/scipp/plopp/issues/342


For the second part, if we can resolve the usage of `fig.show()` above, then we should be able to just show the figrues we want instead of everything.

Reply:
In any case, as all of the above requires more work, we can keep it as png images for now, and always add a new interactive handler later.